### PR TITLE
Resolve package name from output path

### PR DESCRIPTION
**Pull request by bhenderson**
I wanted to be able to run -output . or -output /path/to/dir

Using format.Source was the easiest way I could figure out how to
validate the package name.

Relates to #111

### CODE REVIEW

**Comment by bhenderson** [comment]
travis build broke because it's now using go1.11.5 instead of go1.10.7 and apparently the behavior of go get changed. I'm looking into a fix. I think all it needs is the gnostic run to happen earlier.

File: plugins/gnostic-go-generator/main.go [hunk 1]
```diff
@@ -74,3 +77,15 @@ func main() {
 	err = errors.New("No generated code surface model is available.")
 	env.RespondAndExitIfError(err)
 }
+
+func resolvePackageName(p string) (string, error) {
+	p, err := filepath.Abs(p)
```
Comment (noahdietz) [inline plugins/gnostic-go-generator/main.go]: Can you add some comments in here to describe the resolution behavior?

**Comment by noahdietz** [comment]
Thanks @bhenderson for the PR. Apologies for the slow reply on this. 1 quick comment

File: plugins/gnostic-go-generator/main.go (hunk 1 above)
Comment (bhenderson) [reply plugins/gnostic-go-generator/main.go]: how's that?

File: plugins/gnostic-go-generator/main.go [hunk 2]
```diff
@@ -78,6 +78,8 @@ func main() {
 	env.RespondAndExitIfError(err)
 }
 
+// resolvePackageName converts a path to a valid package name or
+// error if path can't be resolved or resolves to an invalid package name
```
Comment (noahdietz) [inline plugins/gnostic-go-generator/main.go]: sorry for the nit but please add a `.` at the end :)

File: plugins/gnostic-go-generator/main.go (hunk 1 above)
Comment (bhenderson) [reply plugins/gnostic-go-generator/main.go]: I'm not on my normal computer, so I'm just using the GH editor. If you want me to squash the commits I can do that tonight.

Comment (noahdietz) [reply plugins/gnostic-go-generator/main.go]: I'll squash them when I merge, thanks though!

**Pull request merged**